ckanext/dia/model.py

```diff
@@ -0,0 +1,125 @@
+# encoding: utf-8
+
+from __future__ import print_function
+import datetime
+import logging
+import sys
+import six
+from sqlalchemy import Table, Column, types, ForeignKey
+from urllib.parse import urljoin, quote
+from os.path import join
+import uuid
+
+from ckan import model
+from ckan.lib.navl.dictization_functions import validate
+from ckan.lib.navl.validators import not_empty
+from ckan.logic import ValidationError
+from ckan.logic.validators import Invalid
+from ckan.model import DomainObject, User
+from ckan.model.meta import metadata, mapper
+from ckan.plugins import toolkit
+from ckan.common import _, config
+
+
+log = logging.getLogger(__name__)
+minted_uri_table = None
+
+
+def db_setup():
+    if minted_uri_table is None:
+        define_table()
+
+    if not model.package_table.exists():
+        log.critical("Exiting: can not migrate minted uri model \
+if the database does not exist yet")
+        sys.exit(1)
+        return
+
+    if not minted_uri_table.exists():
+        minted_uri_table.create()
+        print("Created Minted URI table")
+    else:
+        print("Minted URI table already exists -- skipping")
+
+
+def define_table():
+    global minted_uri_table
+    if minted_uri_table is not None:
+        return
+    minted_uri_table = Table(
+        'minted_uri', metadata,
+        Column('id', types.Integer, primary_key=True),
+        Column('uri', types.UnicodeText),
+        Column('type', types.UnicodeText),
+        Column('name', types.UnicodeText),
+        Column('created_by_id', types.UnicodeText, ForeignKey('user.id')),
+        Column('created_at', types.DateTime, default=datetime.datetime.utcnow))
+
+    mapper(
+        MintedURI,
+        minted_uri_table
+    )
+
+
+def default_schema():
+    schema = {
+        'type': [not_empty, six.text_type],
+        'name': [not_empty, six.text_type, name_unique],
+        'created_by_id': [not_empty],
+    }
+    return schema
```

Comment:
Does this need to be in a function or could it be a constant?

Reply:
good point, will update to that

Reply:
Function makes more sense with ordering of functions/constants in the file